calls to requestAnimationFrame must check if component has been destroyed

jmathews opened the pull request:
Several of the components (mdTabs, mdBoards, mdMenu, mdStepper, mdListItemExpand) use requestAnimationFrame to monitor scroll position and other attributes of their DOM element.  Since it's possible for a Vue component (and associated elements) to be destroyed before the rAF callback is executed, we set a guard in beforeDestroyed (`this._destroyed = true;`) and abort the callback if it's set.

pablohpsilva commented:
I didn't know vue had that. That's awesome! Thank you for the great fix @jmathews :D

jmathews commented:
Thanks, @pablohpsilva!  Sorry for the noob pull request error.  I updated my fork and rebased my changes against develop.  Hopefully this is the way you want it.

pablohpsilva commented:
Awesoooome!!! :D 
Thank you, my friend.
